setup.py

```diff
@@ -64,14 +78,33 @@ def open_local(paths, mode='r', encoding='utf8'):
     'websockets>=6.0,<7.0',
     'multidict>=4.0,<5.0',
 ]
+tests_require = [
+    'pytest==3.3.2',
+    'multidict>=4.0,<5.0',
+    'gunicorn',
+    'pytest-cov',
+    'aiohttp>=2.3.0,<=3.2.1',
+    'beautifulsoup4',
+    uvloop,
+    ujson,
+    'pytest-sanic',
+    'pytest-sugar'
+]
```

Comment:
There are three requirement files, maybe load requirements from those files would be easier to maintain dependencies of `Sanic`?(cause we only have to focus on requirement files)
BTW, it seems that you change the way to run tests, this might also need to be documented at the [contributing guide](https://github.com/huge-success/sanic/blob/master/CONTRIBUTING.md)

Reply:
@chenjr0719 I like your suggestion. I need to clean up the `requirements-dev.txt` a bit before I can read and use it inside the `setup.py`. Let me get that sorted.

Btw, this doesn't alter the way to run tests. Just provides an extra way to run the test if you get the tarball from `sdist` and you want to run `make test` before you install `sanic` in your machine/environment. This is more of a consumer-facing feature than dev facing change.

Reply:
@chenjr0719 IMHO, I would love it if we can totally get rid of the `requirements*.txt` file and use `extras_require` for additional dependencies and `tests_require` for test related dependencies. 

I am open to suggestion on this matter. I think not having a `requirements*.txt` file is much better option,

Reply:
@harshanarayana To be honest, I would prefer to use `requirements*.txt` files to manage my environment. That's why I propose this idea. But, after taking a look around some projects, use `extras_require` is also a good practice. Hum..., it hard to say which one is better or not cause this might also be influenced by the developer's habit.

Anyway, no matter using `requiements*.txt` files or use `extras_require`, please explain the usage at `README.md` or `CONTRIBUTING.md`.

Reply:
Personally, for running test, i'd prefer tox to create test virtualenv with proper dependencies, or other tools, instead of setup.py/setuptools directly.

Reply:
@chenjr0719 Yes, I am working on adding the necessary details into the Contribution guidelines page. 

@yunstanford The `tox` based testing mechanism will be there. The `setup.py` based test is not a replacement to `tox` based testing setup. `setup.py` based test is just an additional support that can be extended for the consumers of `sanic` who doesn't want to use `whl` files and use the `sdist` tarball instead. They can run the tests and make sure all tests pass before they install it into their environments.

When building/installing something from source, I've found it extremely helpful to have a way to run tests before installing it. It gives me a basic assurance that things will not break abruptly. 
When we build `sdist` we need not package `tox` related configuration with it

Reply:
I like that idea, we just need to make sure to not loose track of this testing ... Perhaps "add a test to test this test" - or _how much wood could a woodchuck chuck if a woodchuck could chuck wood_? :sweat_smile:

Reply:
🆘 Any idea to fix this small annoying issue?
In `sphinx` while using `rst`, if I use either `'` or `"` in the source, when its rendered it's turning it into `‘` and `“`, Any idea how to fix this?

Reply:
@harshanarayana inside a code block or outside? Can you point out the source you're talking about?

Reply:
@vltr Outside the `.. code::` blocks. It's the new section I am trying to add that is causing this problem. 
![image](https://user-images.githubusercontent.com/1404867/50355839-fa4a5a00-0575-11e9-9ca5-e7af60dceef6.png)
![image](https://user-images.githubusercontent.com/1404867/50355856-02a29500-0576-11e9-8736-835389f0ec29.png)

Reply:
Have you tried `\'` or `\"` ?

Reply:
Yes, I tried that. No luck with that either.

Reply:
Hmm ... Why don't you do this instead?
```rst
``pip3 install -e '[.docs]'``
```

Reply:
@vltr 👍 Ah this was a much nicer way to get around the problem. It even makes the document look nicer.